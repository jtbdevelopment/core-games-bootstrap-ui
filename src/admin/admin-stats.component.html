<div class="admin-stats">
    <table class="table table-striped table-sm table-admin-states">
        <tr>
            <th></th>
            <th class="text-right">Last 24 hours</th>
            <th class="text-right">Last 7 days</th>
            <th class="text-right">Last 30 days</th>
            <th class="text-right">All</th>
        </tr>
        <tr class="text-right">
            <th class="text-left">Games</th>
            <td>{{gamesCreated[0]}}</td>
            <td>{{gamesCreated[1]}}</td>
            <td>{{gamesCreated[2]}}</td>
            <td>{{gameCount}}</td>
        </tr>
        <tr class="text-right">
            <th class="text-left">Players Created</th>
            <td>{{playersCreated[0]}}</td>
            <td>{{playersCreated[1]}}</td>
            <td>{{playersCreated[2]}}</td>
            <td>{{playerCount}}</td>
        </tr>
        <tr class="text-right">
            <th class="text-left">Players Logged In</th>
            <td>{{playerLogins[0]}}</td>
            <td>{{playerLogins[1]}}</td>
            <td>{{playerLogins[2]}}</td>
            <td></td>
        </tr>
    </table>
</div>
