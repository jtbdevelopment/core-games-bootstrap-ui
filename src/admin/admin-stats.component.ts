import {Component} from '@angular/core';
import {HttpClient} from '@angular/common/http';

@Component({
    selector: 'admin-stats',
    template: require('./admin-stats.component.html'),
})
export class AdminStatsComponent {
    private static dayInSeconds: number = 86400;

    public playerCount: number = 0;
    public gameCount: number = 0;
    public gamesCreated: number[] = [0, 0, 0];
    public playersCreated: number[] = [0, 0, 0];
    public playerLogins: number[] = [0, 0, 0];

    private time: number = Math.floor((new Date()).getTime() / 1000);
    private times: number[] = [
        (this.time - AdminStatsComponent.dayInSeconds),
        (this.time - (AdminStatsComponent.dayInSeconds * 7)),
        (this.time - (AdminStatsComponent.dayInSeconds * 30))
    ];

    constructor(private http: HttpClient) {
        this.times.forEach((time, index) => {
            this.http.get('/api/player/admin/playersCreated/' + time, {responseType: 'text'})
                .subscribe(text => {
                    this.playersCreated[index] = Number(text);
                });
            this.http.get('/api/player/admin/playersLoggedIn/' + time, {responseType: 'text'})
                .subscribe(text => {
                    this.playerLogins[index] = Number(text);
                });
            this.http.get('/api/player/admin/gamesSince/' + time, {responseType: 'text'})
                .subscribe(text => {
                    this.gamesCreated[index] = Number(text);
                });
        });
        this.http.get('/api/player/admin/gameCount', {responseType: 'text'})
            .subscribe(text => {
                this.gameCount = Number(text);
            });
        this.http.get('/api/player/admin/playerCount', {responseType: 'text'})
            .subscribe(text => {
                this.playerCount = Number(text);
            });
    }
}
