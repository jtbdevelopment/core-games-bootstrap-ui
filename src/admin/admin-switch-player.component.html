<div class="admin-user">
    <div class="row">
        <div class="col-4">
            <button class="btn btn-secondary btn-success btn-stop-simulating"
                    [disabled]="!revertEnabled" (click)="revertToNormal()">Stop
            </button>
        </div>
        <div class="col-8">{{revertText}}</div>
    </div>
    <div class="row">
        <div class="col-4">
            <button class="btn btn-default focus btn-primary btn-get-users"
                    (click)="refreshUsers()">Get Users
            </button>
        </div>
        <div class="col-8">
            <input class="form-control" type="text" [(ngModel)]="searchText"
                   placeholder="And Name contains..">
        </div>
    </div>
    <div class="row">
        <div class="col">
            <table class="table table-striped table-sm table-admin-users">
                <tr>
                    <th>Display Name</th>
                    <th>ID</th>
                    <th>Switch</th>
                </tr>
                <tr *ngFor="let player of players">
                    <td>{{player.displayName}}</td>
                    <td>{{player.id}}</td>
                    <td>
                        <button class="btn btn-danger btn-danger btn-change-user" id="{{player.id}}"
                                (click)="switchToPlayer(player.id)">Switch to this user..
                        </button>
                    </td>
                </tr>
            </table>
        </div>
    </div>
    <div class="row">
        <div class="col">
            <ngb-pagination [pageSize]="pageSize"
                            [directionLinks]="false"
                            [collectionSize]="totalPlayers"
                            [(page)]="currentPage"
                            (pageChange)="changePage()">
            </ngb-pagination>
        </div>
    </div>
</div>
