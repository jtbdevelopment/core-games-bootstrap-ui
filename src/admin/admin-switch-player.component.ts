import {Component} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Player, PlayerService} from 'jtb-core-games-ui';

@Component({
    selector: 'admin-switch-player',
    template: require('./admin-switch-player.component.html'),
})
export class AdminSwitchPlayerComponent {
    public revertText: string = '';
    public revertEnabled: boolean = false;
    public searchText: string = '';
    public players: Player[] = [];

    public pageSize: number = 20;
    public totalPlayers: number = 0;
    public currentPage: number = 1;

    private player: Player;
    private loggedInPlayer: Player;

    constructor(private playerService: PlayerService, private http: HttpClient) {
        this.playerService.player.subscribe(p => {
            this.player = p;
            this.computeRevert();
        });
        this.playerService.loggedInPlayer.subscribe(p => {
            this.loggedInPlayer = p;
            this.computeRevert();
        });
        this.refreshUsers();
    }

    public switchToPlayer(id: string): void {
        this.playerService.simulateUser(id);
    }

    public revertToNormal(): void {
        this.playerService.simulateUser(this.loggedInPlayer.id);
    }

    public changePage(): void {
        this.refreshUsers();
    }

    public refreshUsers(): void {
        this.http.get(
            '/api/player/admin/playersLike',
            {
                params:
                    new HttpParams().set('page', String(this.currentPage - 1)).set('like', this.searchText).set('pageSize', String(this.pageSize))
            }
        ).subscribe(json => {
                this.processUsers(json);
            }
        );
    }

    private processUsers(json: any): void {
//noinspection TypeScriptUnresolvedVariable
        this.totalPlayers = json.totalElements;
        // controller.numberOfPages = response.totalPages;
        let newPlayers = [];
        json.content.forEach(p => {
            newPlayers.push(new Player(p));
        });
        this.players = newPlayers;
        this.currentPage = json.number + 1;
    }

    private computeRevert(): void {
        this.revertEnabled = (this.player && this.loggedInPlayer && this.player.id !== this.loggedInPlayer.id);
        this.revertText = this.revertEnabled ?
            'You are simulating another player.' :
            'You are yourself.';
    }
}
