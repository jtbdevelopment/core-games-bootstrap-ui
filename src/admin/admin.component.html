<div class="admin-screen">
    <ngb-tabset type="pills">
        <ngb-tab title="Stats">
            <ng-template ngbTabContent>
                <admin-stats></admin-stats>
            </ng-template>
        </ngb-tab>
        <ngb-tab title="Switch player">
            <ng-template ngbTabContent>
                <admin-switch-player></admin-switch-player>
            </ng-template>
        </ngb-tab>
    </ngb-tabset>
</div>