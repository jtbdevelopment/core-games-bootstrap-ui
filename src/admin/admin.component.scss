.admin-screen {
  margin-top: 5px;

  button {
    width: 100%;
  }

  table {
    margin-top: 20px;
  }
}