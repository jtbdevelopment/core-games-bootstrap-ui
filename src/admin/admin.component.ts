import {Component} from '@angular/core';

@Component({
    selector: 'admin-screen',
    template: require('./admin.component.html'),
    styles: [require('./admin.component.scss').toString()]
})
export class AdminComponent {

}
