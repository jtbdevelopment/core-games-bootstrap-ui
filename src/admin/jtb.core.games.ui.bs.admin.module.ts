import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {AdminComponent} from './admin.component';
import {AdminStatsComponent} from './admin-stats.component';
import {AdminSwitchPlayerComponent} from './admin-switch-player.component';
import {FormsModule} from '@angular/forms';
import {HttpClientModule} from '@angular/common/http';
import {JTBCoreGamesUI} from 'jtb-core-games-ui';

@NgModule({
    imports: [
        BrowserModule,
        NgbModule,
        FormsModule,
        HttpClientModule,
        JTBCoreGamesUI
    ],
    exports: [
        AdminComponent
    ],
    declarations: [
        AdminComponent,
        AdminStatsComponent,
        AdminSwitchPlayerComponent
    ]
})
export class JTBCoreGamesUIBSAdminModule {
}


