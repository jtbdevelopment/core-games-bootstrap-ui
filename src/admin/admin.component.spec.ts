import {TestBed} from '@angular/core/testing';
import {Component, Input} from '@angular/core';
import {AdminComponent} from './admin.component';
import {NgbModule, NgbTabsetConfig} from '@ng-bootstrap/ng-bootstrap';

@Component({
    selector: 'admin-stats',
    template: '<div class="admin-stats">stats</div>'
})
export class MockAdminStatsComponent {
    @Input() playerLoaded: boolean;
}

@Component({
    selector: 'admin-switch-player',
    template: '<div class="admin-switch-player">switch</div>'
})
export class MockAdminSwitchPlayerComponent {
    @Input() playerLoaded: boolean;
}


describe('admin component', () => {
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbModule
            ],
            declarations: [
                AdminComponent,
                MockAdminStatsComponent,
                MockAdminSwitchPlayerComponent,
            ],
            providers: [
                NgbTabsetConfig
            ],
        });
        TestBed.compileComponents();
    });

    it('should render basics, switch hidden by tabs', () => {
        const fixture = TestBed.createComponent(AdminComponent);
        fixture.detectChanges();
        expect(fixture.nativeElement.querySelector('.admin-stats').textContent.trim()).toEqual('stats');
        expect(fixture.nativeElement.querySelector('.admin-switch-player')).toBeNull();
    });
});
