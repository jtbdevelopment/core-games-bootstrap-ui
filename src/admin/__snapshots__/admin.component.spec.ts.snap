// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin component should render basics, switch hidden by tabs 1`] = `
<admin-screen>
  <div
    class="admin-screen"
  >
    
    
    <ngb-tabset
      ng-reflect-type="pills"
      type="pills"
    >
      
      
      <ul
        class="nav nav-pills justify-content-start"
        role="tablist"
      >
        
        
        
        <li
          class="nav-item"
        >
          
          
          <a
            aria-controls="ngb-tab-0-panel"
            aria-disabled="false"
            aria-expanded="true"
            class="nav-link active"
            href=""
            id="ngb-tab-0"
            role="tab"
          >
            
            Stats
            
            
          
          </a>
          
        
        </li>
        <li
          class="nav-item"
        >
          
          
          <a
            aria-disabled="false"
            aria-expanded="false"
            class="nav-link"
            href=""
            id="ngb-tab-1"
            role="tab"
          >
            
            Switch player
            
            
          
          </a>
          
        
        </li>
        
      
      </ul>
      
      
      <div
        class="tab-content"
      >
        
        
        
        
          
        
        <div
          aria-expanded="true"
          aria-labelledby="ngb-tab-0"
          class="tab-pane active"
          id="ngb-tab-0-panel"
          role="tabpanel"
        >
          
            
          
          
          
          <admin-stats>
            <div
              class="admin-stats"
            >
              stats
            </div>
          </admin-stats>
          
        
          
          
        </div>
        
        
        
          
        
        
        
        
      
      </div>
      
    
    </ngb-tabset>
    
  
  </div>
  
</admin-screen>
`;
